cmd/neofs-lens/internal/storage/sanity.go

```diff
@@ -0,0 +1,232 @@
+	var shards []storageShard
+	defer func() {
+		for _, sh := range shards {
+			_ = sh.m.Close()
+			_ = sh.p.Close()
+			_ = sh.fsT.Close()
+		}
+	}()
+
+	appCfg := config.New(config.Prm{}, config.WithConfigFile(vConfig))
+	err := engineconfig.IterateShards(appCfg, false, func(sc *shardconfig.Config) error {
+		var sh storageShard
+
+		blobStorCfg := sc.BlobStor()
+		metaCfg := sc.Metabase()
+
+		sh.m = meta.New(
+			meta.WithPath(metaCfg.Path()),
+			meta.WithPermissions(metaCfg.BoltDB().Perm()),
+			meta.WithMaxBatchSize(metaCfg.BoltDB().MaxBatchSize()),
+			meta.WithMaxBatchDelay(metaCfg.BoltDB().MaxBatchDelay()),
+			meta.WithBoltDBOptions(&bbolt.Options{Timeout: time.Second}),
+			meta.WithLogger(zap.NewNop()),
+			meta.WithEpochState(epochState{}),
+		)
+
+		for _, subCfg := range blobStorCfg.Storages() {
+			switch subCfg.Type() {
+			default:
+				return fmt.Errorf("unsupported sub-storage type '%s'", subCfg.Type())
+			case peapod.Type:
+				peapodCfg := peapodconfig.From((*config.Config)(subCfg))
+				sh.p = peapod.New(subCfg.Path(), subCfg.Perm(), peapodCfg.FlushInterval())
+
+				var compressCfg compression.Config
+				err := compressCfg.Init()
+				common.ExitOnErr(cmd, common.Errf("failed to init compression config: %w", err))
+
+				sh.p.SetCompressor(&compressCfg)
+			case fstree.Type:
+				fstreeCfg := fstreeconfig.From((*config.Config)(subCfg))
+				sh.fsT = fstree.New(
+					fstree.WithPath(subCfg.Path()),
+					fstree.WithPerm(subCfg.Perm()),
+					fstree.WithDepth(fstreeCfg.Depth()),
+					fstree.WithNoSync(fstreeCfg.NoSync()),
+				)
+			}
+		}
+
+		common.ExitOnErr(cmd, common.Errf("open metabase: %w", sh.m.Open(true)))
+		common.ExitOnErr(cmd, common.Errf("open peapod: %w", sh.p.Open(true)))
+		common.ExitOnErr(cmd, common.Errf("open fstree: %w", sh.fsT.Open(true)))
+
+		// metabase.Open(true) does not set it mode to RO somehow
+		common.ExitOnErr(cmd, common.Errf("moving metabase in readonly mode", sh.m.SetMode(mode.ReadOnly)))
+
+		common.ExitOnErr(cmd, common.Errf("init metabase: %w", sh.m.Init()))
+		common.ExitOnErr(cmd, common.Errf("init peapod: %w", sh.p.Init()))
+		common.ExitOnErr(cmd, common.Errf("init fstree: %w", sh.fsT.Init()))
+
+		shards = append(shards, sh)
+
+		return nil
+	})
+	common.ExitOnErr(cmd, common.Errf("reading config: %w", err))
+
+	for _, sh := range shards {
+		idRaw, err := sh.m.ReadShardID()
+		if err != nil {
+			cmd.Printf("reading shard id: %s; skip this shard\n", err)
+			continue
+		}
+
+		id := base58.Encode(idRaw)
+		cmd.Printf("Checking %s shard\n", id)
+
+		objsChecked, err := checkShard(cmd, sh)
+		if err != nil {
+			if errors.Is(err, context.Canceled) {
+				return
+			}
+
+			cmd.Printf("%d objects checked in %s shard, interrupted by error: %s\n", objsChecked, id, err)
+			continue
+		}
+
+		cmd.Printf("Checked objects in %s shard: %d", id, objsChecked)
+	}
+}
+
+func checkShard(cmd *cobra.Command, sh storageShard) (int, error) {
+	var objectsChecked int
```

Comment:
never updated?

Reply:
fixed. btw, not sure it should be incremented on every object found in metabase or after a successful check only. did success counter